core/src/wallet/bitcoin/explorers/LedgerApiBitcoinLikeBlockchainExplorer.hpp

```diff
@@ -31,42 +31,58 @@
 #ifndef LEDGER_CORE_LEDGERAPIBITCOINLIKEBLOCKCHAINEXPLORER_HPP
 #define LEDGER_CORE_LEDGERAPIBITCOINLIKEBLOCKCHAINEXPLORER_HPP
 
-#include "BitcoinLikeBlockchainExplorer.hpp"
 
+#include <wallet/common/explorers/AbstractLedgerApiBlockchainExplorer.h>
+#include <wallet/bitcoin/explorers/BitcoinLikeBlockchainExplorer.hpp>
 #include <api/BitcoinLikeNetworkParameters.hpp>
 #include <collections/collections.hpp>
 #include <async/Future.hpp>
 #include <net/HttpClient.hpp>
 #include <async/DedicatedContext.hpp>
+#include "api/TransactionParser.hpp"
+#include "api/TransactionsBulkParser.hpp"
+#include "api/BlockParser.hpp"
+#include <api/BitcoinLikeNetworkParameters.hpp>
 
 namespace ledger {
     namespace core {
-        class LedgerApiBitcoinLikeBlockchainExplorer : public BitcoinLikeBlockchainExplorer, DedicatedContext, public std::enable_shared_from_this<LedgerApiBitcoinLikeBlockchainExplorer> {
+
+        using LedgerApiBlockchainExplorer = AbstractLedgerApiBlockchainExplorer<BitcoinLikeBlockchainExplorerTransaction, BitcoinLikeBlockchainExplorer::TransactionsBulk, TransactionsParser, TransactionsBulkParser, BlockParser, api::BitcoinLikeNetworkParameters>;
+        class LedgerApiBitcoinLikeBlockchainExplorer : public BitcoinLikeBlockchainExplorer,
+                                                       public LedgerApiBlockchainExplorer,
+                                                       public DedicatedContext,
```

Comment:
You introduce `public` for `DedicatedContext`; is that on purpose? I don’t mind the change (I think private inheritance is an horrible idea and shouldn’t be used), but maybe you didn’t mean to remove it?

Reply:
No concrete reason, just avoiding the private one and keep coherence with what have been done for other ('similar') classes